### DISCUSSION TITLE
Retain data between requests.

### DISCUSSION THREAD

**Discussion by bhulsman** [livewire/livewire#4394] [1. Help] [answered]
I'm really going crazy and I'm sure I got something similar like this working before... but it just doesn't work. Hopefully someone here can point me in the right direction, because I'm lost!

Let's assume the following:

```
class DateRangeForm extends Component
{
    public string $start_date;

    public string $end_date;

    protected function rules(): array
    {
        return [
            'start_date' => [
                'required',
                'string',
                'date',
            ],
            'end_date' => [
                'required',
                'string',
                'date',
            ],
        ];
    }

    public function mount()
    {
        $this->start_date = '';
        $this->end_date = '';
    }

    public function render()
    {
        $days = [];

        dd($this->start_date, $this->end_date);

        // figure out which days are between the start and end date

        return view('date-range-form', [
            'days' => $days,
        ]);
    }
}
```

together with

```
<input wire:model.lazy="start_date"  type="date" />
<input wire:model.lazy="end_date"  type="date" />
```

1. I comment the `dd`, open the page where the component is displayed.
2. I uncomment the `dd` and select a `start_date`, the `dd` shows the `start_date` and an empty `end_date`.
3. I click away the modal the `dd` output is shown in.
4. I select an `end_date`, but now `start_date` is empty and `end_date` is shown.

What am I doing wrong? I would expect the `start_date` to remain between requests.

**Comment by jonneroelofs** [comment, marked as answer]
Actually it sounds like the dd might be causing the problem. Maybe just render the variables in the form or use the Laravel debug bar to inspect them?

**Comment by Martleby** [comment]
What's going on in your code that does the:

`// figure out which days are between the start and end date`

section?

**Reply by bhulsman** [reply]
I created a CarbonPeriod and filtered out saturdays and sundays to get a list of week days in that period.

**Comment by joshhanley** [comment]
@bhulsman yeah @jonneroelofs is correct the `dd()` stops execution (dump and **die**), which means Livewire won't send any data with the request in response.

So just change that to a `dump()` or change it so it outputs to the log or Ray or something.

Hope this helps!

**Reply by bhulsman** [reply]
That explains a lot! I was just on it to much to even see this. Thanks!

**Reply by joshhanley** [reply]
@bhulsman no worries! Yeah it happens 🙂
